testlib/src/test/java/com/diffplug/spotless/JvmTest.java

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright 2021 DiffPlug
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.diffplug.spotless;
+
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertThrows;
+import static org.junit.Assert.fail;
+
+import java.util.Arrays;
+import java.util.function.Consumer;
+
+import org.junit.Before;
+import org.junit.Test;
+
+public class JvmTest {
+
+	private static final String TEST_NAME = "My Test Formatter";
+	private Jvm.Support<String> testSupport;
+
+	@Before
+	public void initialize() {
+		testSupport = Jvm.support(TEST_NAME);
+	}
+
+	@Test
+	public void supportAddConfig() {
+		Integer differentVersions[] = {0, 1, 2};
+		Arrays.asList(differentVersions).stream().forEach(v -> testSupport.add(v + Jvm.version(), v.toString()));
+		Arrays.asList(differentVersions).stream().forEach(v -> assertThat(testSupport.toString(), containsString(String.format("Version %d", v))));
+		assertThat(testSupport.toString(), containsString(String.format("%s alternatives", TEST_NAME)));
+		testSupport.verifyConfiguration();
+	}
+
+	@Test
+	public void supportConfigChecks() {
+		assertThrows(IllegalArgumentException.class, () -> {
+			testSupport.add(1, "Not a semantic version");
+		});
+
+		testSupport.add(1, "0.1").add(2, "1").verifyConfiguration(); //correct behavior
+		Arrays.<Consumer<Jvm.Support<String>>> asList(
+				s -> {
+					s.add(1, "0.1").add(1, "1.0");
+				}, //forgot to adapt JVM version
+				s -> {
+					s.add(1, "0.1").add(2, "0.1");
+				}, //forgot to adapt formatter version
+				s -> {
+					s.add(1, "1.0").add(2, "0.1");
+				}, //higher formatter version requires lower Java version
+				s -> {
+					s.add(2, "0.1").add(1, "1.0");
+				} //lower formatter version requires higher Java version
+		).stream().forEach(configuration -> {
+			Jvm.Support<String> support = Jvm.support(TEST_NAME);
+			configuration.accept(support);
+			assertThrows(AssertionError.class, () -> {
+				support.verifyConfiguration();
+			});
+		});
+	}
```

Comment:
Will be refactored once #900 is merged.

Reply:
I think #900 is great, and I'm sorry that other PRs are causing merge conflicts for it, but I don't think we should hold other PRs up for it.